Precompilation broken on nightly

I'm unsure if this is an issue with Cthulhu or with Julia, but I find the following on Julia nightly:
```julia
julia> using Cthulhu
[ Info: Precompiling Cthulhu [f68482b8-f384-11e8-15f7-abe071a5a75f]
ERROR: LoadError: AssertionError: src isa OptimizedSource || isnothing(src)
Stacktrace:
  [1] inlining_policy(interp::Cthulhu.CthulhuInterpreter, src::Any, info::Core.Compiler.CallInfo, stmt_flag::UInt8, mi::Core.MethodInstance, argtypes::Vector{Any})
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/interpreter.jl:178
  [2] handle_any_const_result!(cases::Vector{Core.Compiler.InliningCase}, result::Any, match::Core.MethodMatch, argtypes::Vector{Any}, info::Core.Compiler.CallInfo, flag::UInt8, state::Core.Compiler.InliningState{Core.Compiler.EdgeTracker, Core.Compiler.WorldView{Cthulhu.CthulhuCache}, Cthulhu.CthulhuInterpreter}; allow_abstract::Bool, allow_typevars::Bool)
    @ Core.Compiler ./compiler/ssair/inlining.jl:1348
  [3] handle_any_const_result!
    @ ./compiler/ssair/inlining.jl:1340 [inlined]
  [4] compute_inlining_cases(info::Core.Compiler.CallInfo, flag::UInt8, sig::Core.Compiler.Signature, state::Core.Compiler.InliningState{Core.Compiler.EdgeTracker, Core.Compiler.WorldView{Cthulhu.CthulhuCache}, Cthulhu.CthulhuInterpreter})
    @ Core.Compiler ./compiler/ssair/inlining.jl:1433
  [5] handle_call!
    @ ./compiler/ssair/inlining.jl:1478 [inlined]
  [6] assemble_inline_todo!(ir::Core.Compiler.IRCode, state::Core.Compiler.InliningState{Core.Compiler.EdgeTracker, Core.Compiler.WorldView{Cthulhu.CthulhuCache}, Cthulhu.CthulhuInterpreter})
    @ Core.Compiler ./compiler/ssair/inlining.jl:1700
  [7] ssa_inlining_pass!(ir::Core.Compiler.IRCode, state::Core.Compiler.InliningState{Core.Compiler.EdgeTracker, Core.Compiler.WorldView{Cthulhu.CthulhuCache}, Cthulhu.CthulhuInterpreter}, propagate_inbounds::Bool)
    @ Core.Compiler ./compiler/ssair/inlining.jl:103
  [8] run_passes(ci::Core.CodeInfo, sv::Core.Compiler.OptimizationState, caller::Core.Compiler.InferenceResult, optimize_until::Nothing)
    @ Core.Compiler ./compiler/optimize.jl:539
  [9] run_passes
    @ ./compiler/optimize.jl:554 [inlined]
 [10] optimize
    @ ./compiler/optimize.jl:503 [inlined]
 [11] _typeinf(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:271
 [12] typeinf(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:215
 [13] typeinf_edge(interp::Cthulhu.CthulhuInterpreter, method::Method, atype::Any, sparams::Core.SimpleVector, caller::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:952
 [14] abstract_call_method(interp::Cthulhu.CthulhuInterpreter, method::Method, sig::Any, sparams::Core.SimpleVector, hardlimit::Bool, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:609
 [15] abstract_call_gf_by_type(interp::Cthulhu.CthulhuInterpreter, f::Any, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, atype::Any, sv::Core.Compiler.InferenceState, max_methods::Int64)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:153
 [16] abstract_call_known(interp::Cthulhu.CthulhuInterpreter, f::Any, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState, max_methods::Int64)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1829
 [17] abstract_call(interp::Cthulhu.CthulhuInterpreter, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState, max_methods::Nothing)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1900
 [18] abstract_call(interp::Cthulhu.CthulhuInterpreter, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1879
 [19] abstract_eval_statement_expr(interp::Cthulhu.CthulhuInterpreter, e::Expr, vtypes::Vector{Core.Compiler.VarState}, sv::Core.Compiler.InferenceState, mi::Nothing)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2036
 [20] abstract_eval_statement(interp::Cthulhu.CthulhuInterpreter, e::Any, vtypes::Vector{Core.Compiler.VarState}, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2251
 [21] abstract_eval_basic_statement(interp::Cthulhu.CthulhuInterpreter, stmt::Any, pc_vartable::Vector{Core.Compiler.VarState}, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2428
 [22] typeinf_local(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2627
 [23] typeinf_nocycle(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2733
 [24] _typeinf(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:244
 [25] typeinf(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:215
 [26] typeinf_edge(interp::Cthulhu.CthulhuInterpreter, method::Method, atype::Any, sparams::Core.SimpleVector, caller::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:952
 [27] abstract_call_method(interp::Cthulhu.CthulhuInterpreter, method::Method, sig::Any, sparams::Core.SimpleVector, hardlimit::Bool, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:609
 [28] abstract_call_gf_by_type(interp::Cthulhu.CthulhuInterpreter, f::Any, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, atype::Any, sv::Core.Compiler.InferenceState, max_methods::Int64)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:153
 [29] abstract_call_known(interp::Cthulhu.CthulhuInterpreter, f::Any, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState, max_methods::Int64)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1829
 [30] abstract_call(interp::Cthulhu.CthulhuInterpreter, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState, max_methods::Nothing)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1900
 [31] abstract_call(interp::Cthulhu.CthulhuInterpreter, arginfo::Core.Compiler.ArgInfo, si::Core.Compiler.StmtInfo, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1879
 [32] abstract_eval_statement_expr(interp::Cthulhu.CthulhuInterpreter, e::Expr, vtypes::Vector{Core.Compiler.VarState}, sv::Core.Compiler.InferenceState, mi::Nothing)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2036
 [33] abstract_eval_statement(interp::Cthulhu.CthulhuInterpreter, e::Any, vtypes::Vector{Core.Compiler.VarState}, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2251
 [34] abstract_eval_basic_statement(interp::Cthulhu.CthulhuInterpreter, stmt::Any, pc_vartable::Vector{Core.Compiler.VarState}, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2452
 [35] typeinf_local(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2627
 [36] typeinf_nocycle(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:2733
 [37] _typeinf(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:244
 [38] typeinf(interp::Cthulhu.CthulhuInterpreter, frame::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/typeinfer.jl:215
 [39] do_typeinf!(interp::Cthulhu.CthulhuInterpreter, mi::Core.MethodInstance)
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:707
 [40] mkinterp(interp::Core.Compiler.NativeInterpreter, args::Any)
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:724
 [41] _descend(term::REPL.Terminals.TTYTerminal, args::Any; interp::Core.Compiler.NativeInterpreter, kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:iswarn,), Tuple{Bool}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:746
 [42] kwcall(::NamedTuple{(:iswarn,), Tuple{Bool}}, ::typeof(Cthulhu._descend), term::REPL.Terminals.TTYTerminal, args::Any)
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:744
 [43] __descend_with_error_handling(args::Any; terminal::Any, kwargs::Base.Pairs{Symbol, V, Tuple{Vararg{Symbol, N}}, NamedTuple{names, T}} where {V, N, names, T<:Tuple{Vararg{Any, N}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:234
 [44] kwcall(::Any, ::typeof(Cthulhu.__descend_with_error_handling), args::Any)
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:231
 [45] _descend_with_error_handling(f::Any, argtypes::Any; kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:iswarn, :terminal), Tuple{Bool, REPL.Terminals.TTYTerminal}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:223
 [46] kwcall(::NamedTuple{(:iswarn, :terminal), Tuple{Bool, REPL.Terminals.TTYTerminal}}, ::typeof(Cthulhu._descend_with_error_handling), f::Any, argtypes::Any)
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:215
 [47] descend_code_typed(::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, REPL.Terminals.TTYTerminal, Tuple{Symbol}, NamedTuple{(:terminal,), Tuple{REPL.Terminals.TTYTerminal}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:168
 [48] kwcall(::NamedTuple{(:terminal,), Tuple{REPL.Terminals.TTYTerminal}}, ::typeof(Cthulhu.descend_code_typed), ::Any, ::Any)
    @ Cthulhu ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:168
 [49] top-level scope
    @ ~/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:815
 [50] include
    @ ./Base.jl:450 [inlined]
 [51] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1649
in expression starting at /home/jishnu/.julia/packages/Cthulhu/qdhmj/src/Cthulhu.jl:1
in expression starting at stdin:1
ERROR: Failed to precompile Cthulhu [f68482b8-f384-11e8-15f7-abe071a5a75f] to /home/jishnu/.julia/compiled/v1.9/Cthulhu/jl_FMLuHF.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, keep_loaded_modules::Bool)
   @ Base ./loading.jl:1802
 [3] compilecache
   @ ./loading.jl:1746 [inlined]
 [4] _require(pkg::Base.PkgId, env::String)
   @ Base ./loading.jl:1432
 [5] _require_prelocked(uuidkey::Base.PkgId, env::String)
   @ Base ./loading.jl:1296
 [6] macro expansion
   @ ./loading.jl:1276 [inlined]
 [7] macro expansion
   @ ./lock.jl:267 [inlined]
 [8] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1239

(@v1.9) pkg> st Cthulhu
Status `~/.julia/environments/v1.9/Project.toml`
  [f68482b8] Cthulhu v2.7.5

julia> VERSION
v"1.9.0-DEV.1599"
```

--- a/src/interpreter.jl
+++ b/src/interpreter.jl
@@ -175,7 +175,6 @@ end
 @static if isdefined(CC, :CallInfo)
 function CC.inlining_policy(interp::CthulhuInterpreter,
     @nospecialize(src), @nospecialize(info::CCCallInfo), stmt_flag::UInt8, mi::MethodInstance, argtypes::Vector{Any})
-    @assert isa(src, OptimizedSource) || isnothing(src)
     if isa(src, OptimizedSource)
         if CC.is_stmt_inline(stmt_flag) || src.isinlineable
             return src.ir

